mlir/include/mlir/Dialect/MemRef/IR/MemRefOps.td

```diff
@@ -1881,7 +1885,11 @@ def MemRef_StoreOp : MemRef_Op<"store",
     The `store` op stores an element into a memref at the specified indices.
 
     The number of indices must match the rank of the memref. The indices must
-    be in-bounds: `0 <= idx < dim_size`
+    be in-bounds: `0 <= idx < dim_size`.
+
+    Lowerings of `memref.store` may emit annotations that would cause undefined
+    behavior if indices are out of bounds or if computing the offset in the
+    memref would cause signed overflow of the `idx` type.
```

Comment:
Nit: "\`index\`" type, since these indices are always of type `index`

Reply:
Thanks, done!